### PR TITLE
Added support for "contains" operator on conditional display

### PR DESCRIPTION
**Pull request by georgepemberton1998**
### Description
show_when can now be passed on a component or page with the `contains` operator. This is different to the `in` operator in that the in operator is used to pass an array of accepted values that a given field is "in", whereas `contains` is used to pass a single accepted value which is "in" a field.

### To test
Covered by accompanying unit tests. Functionality can be tested in the Form Renderer Sandbox by adding a component with a show_when with "op": "contains".
1. Add an array field to the `initial data` tab in form renderer sandbox. e.g. "something" : ["alpha", "bravo", "charlie" ]
2. use the below component and confirm that it is displayed
```
    {
      "id": "status",
      "fieldId": "status",
      "type": "text",
      "label": "select status",
      "required": true,
      "show_when": [
        {
          "field": "something",
          "op": "contains",
          "value": "alpha"
        }
      ]
    }
```
3. change the component check for a value not in the array and confirm that it is no longer displayed
```
    {
      "id": "status",
      "fieldId": "status",
      "type": "text",
      "label": "select status",
      "required": true,
      "show_when": [
        {
          "field": "something",
          "op": "contains",
          "value": "delta"
        }
      ]
    }
```

### CODE REVIEW

**Comment by georgepemberton1998** [comment]
Converting to a draft until PR #163 has been merged

File: src/utils/Condition/meetsCondition.js [hunk 1]
```diff
@@ -45,6 +45,9 @@ const meetsCondition = (condition, value) => {
         // If it's not an array, nothing can be IN it, so it must meet the condition.
         return true;
       }
+      case 'contains': {
+        return value.toString().includes(compare);
```
Comment (paulfgraham) [inline src/utils/Condition/meetsCondition.js:49]: If `value` is `null` or `undefined`, this will fail. It's also case-sensitive, which _may_ be the desired behaviour but may actually just be something that hasn't been considered. If the former, can we have a note expressing why case sensitivity is important.

File: src/docs/ConditionalDisplayExamples.stories.mdx [hunk 2]
```diff
@@ -154,3 +154,28 @@ if the value of `areYouACivilServant` is "yes", `delegateEmails` will still be s
...
+    "values": "BMW"
+  }
+}
+```
+
+In the above example, the `BMWModel` component **will** be shown when the value
+of the `cars` field **contains** `BMW`.
+
+In this scenario, the component will be shown if the `cars` field is one of the following: <br/>
+An array containing `BMW` e.g. `"cars": "['Ford', 'Mercedes', 'BMW', 'Lexus']"` <br/>
```
Comment (paulfgraham) [inline src/docs/ConditionalDisplayExamples.stories.mdx:176]: You don't need this for arrays unless you want to consider sub-strings of the items in arrays. For arrays, `op: 'in'` does that job.

Also, in your example, `cars` is _not_ an array, it's a string representation of an array.

File: src/utils/Condition/meetsCondition.test.js [hunk 3]
```diff
@@ -269,6 +269,69 @@ describe('utils.Condition.meetsCondition', () => {
...
+      const CONDITION = getCondition(op, VALUE)
+      expect(meetsCondition(CONDITION, FIELD)).toBeFalsy();
+    });
+    it('should reject any value when the field is an empty string', () => {
+      const FIELD = '';
+      const VALUE = 'alpha';
+      const CONDITION = getCondition(op, VALUE)
+      expect(meetsCondition(CONDITION, FIELD)).toBeFalsy();
+    });
+
```
Comment (paulfgraham) [inline src/utils/Condition/meetsCondition.test.js:332]: As mentioned in a previous comment, you need tests for `null` and `undefined` values.

**Review by paulfgraham — CHANGES_REQUESTED** [review]

File: src/docs/ConditionalDisplayExamples.stories.mdx (hunk 2 above)
Comment (georgepemberton1998) [reply src/docs/ConditionalDisplayExamples.stories.mdx:176]: My understand with `op: 'in'` is that it can be used to provide an `array value` and check if a `field` is "in" that array value. The `contains` operator would essentially do the opposite of this. You can provide a `field array` and check if a given value is in that. I looked at uplifting the "in" operator to enable this functionality but I believed it to be quite confusing.

To clarify - as part of the context data we have an array field "groups" (below) - I would like to display a component if `"/bf-intel"` is in the `groups` array. I was unable to see a way to use this with the "in" operator without a code change to it. Please let me know if I have missed something.

```
"shiftDetailsContext": {
    [
     ---
    "groups" [
      "/bf-intel",
      "/bfrole",
      "/camunda-admin"
    ]
    ---
    ]
}
```

Comment (paulfgraham) [reply src/docs/ConditionalDisplayExamples.stories.mdx:176]: Ah, I see what you mean. Yes, you're absolutely right!

The point about it being a string representation of an array remains, though. 🙂

File: src/utils/Condition/meetsCondition.js [hunk 4]
```diff
@@ -46,7 +46,11 @@ const meetsCondition = (condition, value) => {
         return true;
       }
       case 'contains': {
-        return value.toString().includes(compare);
+        if (value){
```
Comment (paulfgraham) [inline src/utils/Condition/meetsCondition.js:49]:
```suggestion
        if (value) {
```

Alternatively, this whole thing could simply be:
```
        return value?.toString().includes(compare);
```

File: src/utils/Condition/meetsCondition.js [hunk 5]
```diff
@@ -46,7 +46,11 @@ const meetsCondition = (condition, value) => {
...
+          return value.toString().includes(compare);
+        }
+       // If no value is provided, the field cannot contain it, so it must fail the condition.
```
Comment (paulfgraham) [inline src/utils/Condition/meetsCondition.js:52]: The indenting on this line and the one below is off - it's a single space too far back.

File: src/utils/Condition/meetsCondition.test.js [hunk 6]
```diff
@@ -329,6 +329,30 @@ describe('utils.Condition.meetsCondition', () => {
...
+      const CONDITION = getCondition(op, VALUE)
+      expect(meetsCondition(CONDITION, FIELD)).toBeFalsy();
+    });
+    it('should reject any field when the value is undefined', () => {
+      const FIELD = ['alpha','bravo','charlie'];
+      const VALUE = undefined;
+      const CONDITION = getCondition(op, VALUE)
+      expect(meetsCondition(CONDITION, FIELD)).toBeFalsy();
+    });
+    it('should reject any value when the field is undefined', () => {
```
Comment (paulfgraham) [inline src/utils/Condition/meetsCondition.test.js:344]:
```suggestion
    it('should reject any value when the field is null', () => {
```

**Review by paulfgraham — CHANGES_REQUESTED** [review]
